File: solar/public/js/cotizacion.py
# Copyright (c) 2020, KIRE INFORMATICA and contributors
# For license information, please see license.txt

from __future__ import unicode_literals
import frappe
from frappe.model.document import Document

#class Quotation(Document):
#	def solar(self):
		# frappe.throw('Desarrollo en proceso')
#		crear_ingenieria(self)

#@frappe.whitelist()
#def total_proyectado():
#	cotizador = frappe.db.get_list('Cotizador',fields=['name'])

#	for c in cotizador:
#		tarifas = frappe.db.get_list('tarifa', filters={ 'parent': c.name,'idx':['=', '1'] },
#	    fields=['idx', 'parent','precio'],
#	    order_by='posting_date asc',
#	    page_length=1,
#	    if facturas:
#			row = tarifas[0][1][2]
#		    frappe.errprint(c.name, tarifas[0][1][2])
#	return

File: solar/public/js/cotizacion.js
// Copyright (c) 2020, KIRE INFORMATICA and contributors
// For license information, please see license.txt

//frappe.ui.form.on('Quotation', {
//	ingenieria: function(frm) {
//		frm.call('solar')
//}})

frappe.ui.form.on("Opportunity", {
    refresh: function(frm) {
      frm.add_custom_button(__("Ingenieria"), function() {
            frappe.new_doc("Cotizador", {
							  opportunity: frm.doc.name,
                cotizacion_para: frm.doc.enquiry_from,
                prospecto: frm.doc.lead,
                cliente: frm.doc.customer
            });
        });
    }
});

frappe.ui.form.on("Opportunity Item", {
    precio1: function(frm, cdt, cdn) {
        var d = locals[cdt][cdn];
        frappe.model.set_value(d.doctype, d.name, "total1", d.precio1 * d.qty);
        var total = 0
        frm.doc.items.forEach(function(d) {
        total += d.total1;
        });
    frm.set_value('precio_total1', total);
    refresh_field("precio_total1");
    }
});

frappe.ui.form.on("Opportunity Item", {
    qty: function(frm, cdt, cdn) {
        var d = locals[cdt][cdn];
        frappe.model.set_value(d.doctype, d.name, "total1", d.precio1 * d.qty);
        var total = 0
        frm.doc.items.forEach(function(d) {
        total += d.total1;
        });
    frm.set_value('precio_total1', total);
    refresh_field("precio_total1");
    }
});
